cmd/install/assets/hypershift_operator.go

```diff
@@ -103,6 +104,141 @@ func (o HyperShiftOperatorOIDCProviderS3Secret) Build() *corev1.Secret {
 	return secret
 }
 
+type ExternalDNSCredsSecret struct {
+	Namespace  *corev1.Namespace
+	CredsBytes []byte
+}
+
+func (o ExternalDNSCredsSecret) Build() *corev1.Secret {
+	secret := &corev1.Secret{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Secret",
+			APIVersion: corev1.SchemeGroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      externaDNSCredsSecretName,
+			Namespace: o.Namespace.Name,
+		},
+		Data: map[string][]byte{
+			"credentials": o.CredsBytes,
+		},
+	}
+	return secret
+}
+
+type ExternalDNSDeployment struct {
+	Namespace         *corev1.Namespace
+	Image             string
+	ServiceAccount    *corev1.ServiceAccount
+	Provider          string
+	DomainFilter      string
+	CredentialsSecret *corev1.Secret
+}
+
+func (o ExternalDNSDeployment) Build() *appsv1.Deployment {
+	replicas := int32(1)
+	deployment := &appsv1.Deployment{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Deployment",
+			APIVersion: appsv1.SchemeGroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "external-dns",
+			Namespace: o.Namespace.Name,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"name": "external-dns",
+				},
+			},
+			Replicas: &replicas,
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"name":                    "external-dns",
+						"app":                     "external-dns",
+						hyperv1.OperatorComponent: "external-dns",
+					},
+				},
+				Spec: corev1.PodSpec{
+					ServiceAccountName: o.ServiceAccount.Name,
+					Containers: []corev1.Container{
+						{
+							Name:            "external-dns",
+							Image:           o.Image,
+							ImagePullPolicy: corev1.PullIfNotPresent,
+							Command:         []string{"/external-dns"},
+							Args: []string{
+								"--source=service",
+								"--source=openshift-route",
+								fmt.Sprintf("--domain-filter=%s", o.DomainFilter),
+								fmt.Sprintf("--provider=%s", o.Provider),
+								"--aws-zone-type=public",
```

Comment:
Is it ok `--aws-zone-type=` unconditionally regardless of the platform?

Reply:
We can set it in the platform specific section :+1: